packages/syft/src/syft/custom_worker/runner_k8s.py

```diff
@@ -166,67 +147,49 @@ def _create_image_pull_secret(
         reg_url: str,
         **kwargs,
     ):
-        _secret = Secret(
-            {
-                "metadata": {
-                    "name": f"pull-secret-{pool_name}",
-                    "labels": {
-                        "app.kubernetes.io/name": KUBERNETES_NAMESPACE,
-                        "app.kubernetes.io/component": pool_name,
-                        "app.kubernetes.io/managed-by": "kr8s",
-                    },
-                },
-                "type": "kubernetes.io/dockerconfigjson",
-                "data": {
-                    ".dockerconfigjson": self._create_dockerconfig_json(
-                        reg_username,
-                        reg_password,
-                        reg_url,
-                    )
-                },
-            }
+        return KubeUtils.create_dockerconfig_secret(
+            secret_name=f"pull-secret-{pool_name}",
+            component=pool_name,
+            registries=[
+                (reg_url, reg_username, reg_password),
+            ],
         )
 
-        return self._create_or_get(_secret)
-
     def _create_stateful_set(
         self,
         pool_name: str,
         tag: str,
         replicas=1,
-        env_vars: Optional[dict] = None,
+        env_vars: Optional[List[Dict]] = None,
+        mount_secrets: Optional[Dict] = None,
         pull_secret: Optional[Secret] = None,
         **kwargs,
     ) -> StatefulSet:
         """Create a stateful set for a pool"""
 
-        env_vars = env_vars or {}
+        volumes = []
+        volume_mounts = []
         pull_secret_obj = None
-
-        _pod = Pod.get(self._current_pod_name())
-
-        creds_volume = self._get_obj_from_list(
-            objs=_pod.spec.volumes,
-            name="credentials-data",
-        )
-        creds_volume_mount = self._get_obj_from_list(
-            objs=_pod.spec.containers[0].volumeMounts,
-            name="credentials-data",
-        )
-
-        env = _pod.spec.containers[0].env.to_list()
-        env_clone = copy.deepcopy(env)
```

Comment:
Before we were copying volumes and mount points over here, but now i've made this function more generic. The responsibility for worker-specific env provisioning is moved to `service/workers/utils.py`. That way we can tweak around a bunch of things at a higher level in the call stack.